### ISSUE TITLE
Evaluation metrics meaning

### ISSUE THREAD

**Issue by aclifton314** [allenai/allennlp#3090]
**System (please complete the following information):**
 - OS: Linux (virtual machine)
 - Python version: 3.6.1
 - AllenNLP version: v0.8.4
 - PyTorch version: 1.0.1

**Question**
I am using AllenNLP to help build a question answering system. I was able to execute the “allennlp evaluate” command with my chosen model and input file. I was having a difficult time understanding the meaning behind the various metrics that are outputted form this command (start_acc, end_acc, span_acc, em, f1, loss). 

I read the SQuAD paper for em and f1 so I believe I understand those. My questions are:
1) What are the meanings of the other metrics and how are they calculated?
2) What is the difference between span_acc and em?

**Comment by matt-gardner** [comment]
I provided you with this answer on our email list (pretty sure you're the same person):

> For EM / F1, I'd refer you to the SQuAD paper, which defines the metrics that they use (we use their script for calculating those metrics).  Their EM (exact match) metric does some normalization of the predicted span before calculating a match.  `span_acc` is a more strict "exact match", in that it is just exact match of the start and end position of the predicted span with the gold span, without any normalization, and `start_acc` and `end_acc` are the pieces of that.

I thought I answered both of those questions - what part isn't clear?

**Comment by aclifton314** [comment]
Thanks for that. I thought I would post this here as some of my colleagues had similar questions. 

1) Do you happen to know what that normalization is for EM? 
2) Is span_acc, then, how often the predicted answer matches exactly the gold span?

**Comment by matt-gardner** [comment]
1. You can see how EM and F1 are calculated [here](https://github.com/allenai/allennlp/blob/master/allennlp/tools/squad_eval.py) (script copied directly from the creators of SQuAD).
2. `span_acc` is just using standard categorical accuracy metrics on the given training data, and is analogous to the loss computation (both are just functions of the predicted distributions over paragraph indices and the labels, not any strings).  This is more strict than EM in two ways: (1) there is no normalization (removal of determiners, punctuation, etc.), and (2) it is position dependent, while EM is not.  If we predict an identical string in the wrong position, this is an error for `span_acc` but not for EM.

**Comment by aclifton314** [comment]
Got it. Thank you!
